### PR TITLE
Add useful test compilation documentation

### PR DESCRIPTION
**Pull request by blueredpen**
I found this great resource which shed more light on how to compile the tests:
https://jonatack.github.io/articles/how-to-compile-bitcoin-core-and-run-the-tests

### CODE REVIEW

**Comment by brunoerg** [comment]
Concept NACK. The documentation should be enough to compile the tests, I think if a website has useful tips about it, then, we should include the tips in the docs, not a reference to an external website.

**Comment by blueredpen** [comment]
Ok, I'll go ahead and revise the docs with the contents of that website.

**Comment by blueredpen** [comment]
@brunoerg I disagree. The website makes it very clear the step-by-step instructions one needs to follow in order to build Bitcoin from source and then run the unit tests. It contains a lot more detail than the docs in the README.md file and is much easier to follow, in my opinion.

Could we keep the website as a link at the bottom of the page as-is?

**Comment by hebasto** [comment]
Documentation in this repo should be concise and include only required steps.

**Comment by blueredpen** [comment]
That website is super concise and easy-to-follow. I'm not able to compile the tests by following the README.md because it leaves-out key details (for ex. building BerkeleyDB).

**Comment by brunoerg** [comment]
> @brunoerg I disagree. The website makes it very clear the step-by-step instructions one needs to follow in order to build Bitcoin from source and then run the unit tests. It contains a lot more detail than the docs in the README.md file and is much easier to follow, in my opinion.
> 
> Could we keep the website as a link at the bottom of the page as-is?
> 
> @jonatack what do you think about this change to the README.md?

@jonatack's articles are really good. However, `if you're not able to compile the tests successfully yet` sounds like the doc could be not enough for it. So, if the article covers a required step or information that is not covered by the doc, we could add it to the doc instead of putting a reference to the article.

**Comment by blueredpen** [comment]
I think the README.md file needs to be replaced by Jon's article. README.md contains quite a number of very specific points, like adding tests and debugging segmentation faults. I can go back and revise README.md in this way.

**Comment by fanquake** [comment]
Thanks, but we're not going to make these changes. The test README is not the place for instructions on installing dependencies (some of which are incorrect. i.e Protobuf is not required), or compiling Bitcoin Core. The compilation documentation is in the `build-*` docs.

If you're running into an issue compiling, after following our build documentation, feel free to open an issue with steps to reproduce.

**Comment by blueredpen** [comment]
@fanquake I'm unable to run the unit tests by follow the doc as it stands. For example, it makes no mention of running autogen.sh or the directory I should be running the commands in.

That article that I'm referencing gave me everything I need to run the tests successfully. Could we improve the current README.md documents to reflect the necessary details in order to be able to run the tests?

**Comment by fanquake** [comment]
> For example, it makes no mention of running autogen.sh

I'd suggest reading the relevant `build-*.md` documentation for whichever platform you are using, as how to compile the code is already well documented, and in most cases, instructions on how to run the (unit) tests are mentioned. We aren't going to add duplicate dependency installation / compilation instructions to this README.

**Comment by blueredpen** [comment]
Will check it out, thanks.

**Comment by prayank23** [comment]
> I'm unable to run the unit tests by follow the doc as it stands. For example, it makes no mention of running autogen.sh or the directory I should be running the commands in.

> That article that I'm referencing gave me everything I need to run the tests successfully. 

I agree documentation in this repository can be improved and I have used https://jonatack.github.io/articles/how-to-compile-bitcoin-core-and-run-the-tests this for help lot of times.

**Comment by jonatack** [comment]
> some of which are incorrect. i.e Protobuf is not required

@fanquake thanks for the feedback! Indeed, the dependencies section of https://jonatack.github.io/articles/how-to-compile-bitcoin-core-and-run-the-tests was in need of an update. Done in https://github.com/jonatack/jonatack.github.io/commit/057c3aa9bb673f334c292441b83631c3b014664b. Future suggestions/corrections welcome.

File: src/test/README.md [hunk 1]
```diff
@@ -100,3 +131,7 @@ gdb src/test/test_bitcoin core
 
 (gbd) bt  # produce a backtrace for where a segfault occurred
 ```
+
+#### Additional guide
+Below is a great guide on how to run the tests if you're not able to compile the tests successfully yet:
+https://jonatack.github.io/articles/how-to-compile-bitcoin-core-and-run-the-tests
```
Comment (hebasto) [inline src/test/README.md:137]: NACK on adding links to personal sites, blogs etc.

Although, https://jonatack.github.io/articles/how-to-compile-bitcoin-core-and-run-the-tests indeed is a great piece to read for newbies.

**Comment by MarcoFalke** [comment]
Agree with everyone else, if there is something required missing from our docs, it should be added here. The docs here should be both sufficient and self-contained.

**Pull request closed without merging**